Makefile

```diff
@@ -0,0 +1,21 @@
+# generate version string for the build
+VERSION := $(shell hack/get-version.sh)
+
+# set constants
+IMAGE_REGISTRY ?= "hasura"
+CONTROLLER_IMAGE ?= "$(IMAGE_REGISTRY)/gitkube-controller"
+GITKUBED_IMAGE ?= "$(IMAGE_REGISTRY)/gitkubed"
+GITKUBED_DIR ?= "build/gitkubed"
+SETUP_MANIFEST_FILE ?= "manifests/gitkube-setup.yaml"
+
+build-controller:
+	docker build -t $(CONTROLLER_IMAGE):$(VERSION) .
+	docker push $(CONTROLLER_IMAGE):$(VERSION)
+	hack/update-image-version.sh $(SETUP_MANIFEST_FILE) $(CONTROLLER_IMAGE) $(VERSION)
+
+build-gitkubed:
+	docker build -t $(GITKUBED_IMAGE):$(VERSION) $(GITKUBED_DIR)
+	docker push $(GITKUBED_IMAGE):$(VERSION)
```

Comment:
should we push on build?

Reply:
Make push a different rule?

Reply:
Yeah, build and push should be separate because you might build many and push once.